docker/linux-build.dockerfile

```diff
@@ -119,6 +119,11 @@ COPY --from=build-native ${PUBLISH_FOLDER} ${TRACER_HOME}
 COPY --from=build-native /var/log/datadog/ /var/log/datadog/
 
 
+FROM scratch as native-linux-binary
```

Comment:
What is this used for?

Reply:
This is used by: 

https://github.com/DataDog/dd-trace-dotnet/blob/tony/ci-cli-tool/src/Datadog.Trace.Tools.Runner/linux-native-build.cmd#L6

The idea is to have an image with just only the native .so file, and with the `--output type=local,dest=home/linux-x64` from the `docker build` command extract that image on the `home/linux-x64` folder.

Reply:
cool, thanks!

Reply:
Wow that's super cool!

Reply:
I have to change this method to copy the linux binary because it's use the new docker BuildKit engine and while works in my desktop because WSL2, it's failing in the azure pipeline due https://github.com/MicrosoftDocs/azure-devops-docs/issues/6459. I having the same error in the runner pipeline :(

I will not merge this PR until I get this fixed.

Reply:
Well I ran in another problem with Azure, I changed the method used but now:

```bash
2020-10-16T13:56:01.6531647Z           *** Checking for docker daemon...
2020-10-16T13:56:01.6532733Z          
2020-10-16T13:56:10.3175098Z          linux-native-build.cmd
2020-10-16T13:56:10.3761960Z          *** Building Native ClrProfiler for linux-x64 ***
2020-10-16T13:56:20.9911396Z          Sending build context to Docker daemon  60.18MB
2020-10-16T13:56:20.9912343Z          
2020-10-16T13:56:22.3441350Z          Error response from daemon: unsupported os linux
```

The docker for windows in azure cannot build linux images, so I can't create the linux binaries from azure windows vm.

Anyone have any ideas? I'm open to hearing them. @DataDog/apm-dotnet

Reply:
Maybe this can be solved with a custom vm in a private pool?

Reply:
@lucaspimentel @zacharycmontoya

Reply:
I changed the `runner.yaml` pipeline to use two jobs, the first one creates the native linux binary and the second one finishes the build. Also uploads all the artifacts: https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=28235&view=artifacts&type=publishedArtifacts